### DISCUSSION TITLE
CustomBackground callback [python]

### DISCUSSION THREAD

**Discussion by kutraj87** [pthom/hello_imgui#74] [Q&A] [answered]
Firstly, thanks a lot for a wonderful framework pthom! 

I've been using imgui_bundle (python) for an OpenGL based rendering project. I've been trying to port my application to use hello_imgui for incorporating docking splits. I'm trying to do this with an FBO inside a DockableWindow and using imgui::Image(). I'm having issues with the max_content_size inside this window so the OpenGL texture is getting cropped (story for another thread perhaps?) 

For this reason, I found the CustomBackground callback - it looks like it will exactly solve my particular use-case! Instead of rendering to a separate FBO, I can just render to the default framebuffer ("background window"). Please let me know if I've misunderstood this. I have tried using the background draw lists feature in imgui but I keep getting weird clipping issues.

I noticed that the CustomBackground callback is mentioned in the hello_imgui_api.md file. Just curious about your plans for incorporating this into imgui-bundle python bindings.

Thanks again!

**Comment by pthom** [comment, marked as answer]
Hello, and thanks for your kind words!

The CustomBackground callback is brand new (yesterday!) and was not added into ImGui Bundle yet. It was added now: 

see https://pthom.github.io/imgui_bundle/quickstart.html#_custom_3d_background

I hope this helps!

**Comment by kutraj87** [comment]
Many thanks! This is exactly what was needed. Works like a charm :)
